tests/test_edgeql_update.py

```diff
@@ -2349,6 +2353,126 @@ async def test_edgeql_update_subtract_02(self):
             ],
         )
 
+    async def test_edgeql_update_subtract_non_distinct(self):
+        await self.con.execute("""
+            INSERT UpdateTest {
+                name := 'update-test-subtract-non-distinct-1',
+                str_tags := {'1', '1', '2', '3', '2', '2', '3', '4'},
+            };
+            INSERT UpdateTest {
+                name := 'update-test-subtract-non-distinct-2',
+                str_tags := {'1', '2', '2', '3', '4', '5'},
+            };
+        """)
+
+        await self.assert_query_result(
+            r"""
+                SELECT
+                    UpdateTest {
+                        str_tags ORDER BY UpdateTest.str_tags
+                    }
+                FILTER
+                    .name LIKE 'update-test-subtract-non-distinct-%'
+                ORDER BY
+                    .name
+            """,
+            [
+                {
+                    'str_tags': ['1', '1', '2', '2', '2', '3', '3', '4'],
+                },
+                {
+                    'str_tags': ['1', '2', '2', '3', '4', '5'],
+                },
+            ],
+        )
+
+        await self.con.execute("""
+            UPDATE UpdateTest
+            FILTER .name LIKE 'update-test-subtract-non-distinct-%'
+            SET {
+                str_tags -= {'1', '2', '2', '3', '5'}
```

Comment:
Side question, is it legal to use both `-=` and `+=` in the same `SET` for the same property (it should be, I guess)? Can you add tests for that too?

Reply:
> Side question, is it legal to use both `-=` and `+=` in the same `SET` for the same property

It isn't:

```
error: duplicate definition of property 'p' of object type 'default::Foo'
  ┌─ query:1:37
  │
1 │ UPDATE Foo SET { p -= {1, 2, 2, 3}, p += {1} };
```

Reply:
Obviously out of scope of this PR, but I'm curious what you think about it? Should we allow if post 1.0?

Reply:
No.  It would be a complicated thing to support and it potentially opens the door for unintended copy-paste bugs.